lib/stat_tracker.rb

```diff
@@ -87,22 +87,36 @@ def count_of_teams
        @teams_stats_data.size
     end
    
-    def best_offense
-
-        teams_goals = Hash.new(0)
-        games_played = Hash.new(0)
+    def team_seasons_goals(location)
+        
+        teams_data = Hash.new { |hash, key| hash[key] = { goals: 0, games_played: 0 } }
 
         @seasons_stats_data.each do |game_id, season|
-            teams_goals[season.team_id] += season.goals
-            games_played[season.team_id] += 1
+            if location.include?(season.hoa)
+                teams_data[season.team_id][:goals] += season.goals
+                teams_data[season.team_id][:games_played] += 1
+            end
+        end
+        teams_data
+    end
+
+    def teams_scores_average_min_by(teams_data)
+        lowest_scoring_team = teams_data.min_by do |team_id, team_data|
+            team_data[:goals].to_f / team_data[:games_played].to_f
         end
+    end
 
-        best_offense_team = teams_goals.max_by do |team_id, goals|
-            goals.to_f / games_played[team_id].to_f
+    def teams_scores_average_max_by(teams_data)
+        highest_scoring_team = teams_data.max_by do |team_id, team_data|
+            team_data[:goals].to_f / team_data[:games_played].to_f
         end
+    end
 
-        best_offense_team_id = best_offense_team[0]
+    def best_offense
+        teams_goals_data = team_seasons_goals(['home','away'])
+        best_offense_team = teams_scores_average_max_by(teams_goals_data)
```

Comment:
The `teams_scores_average_max_by` helper method will only continue to help streamline our code, thanks for working this out.